packages/contracts/src/contracts/current/protocol/AssetTransferProxy/AssetTransferProxy.sol

```diff
@@ -0,0 +1,116 @@
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "./IAssetTransferProxy.sol";
+import "./IAssetProxy.sol";
+import "../../utils/Authorizable/Authorizable.sol";
+
+contract AssetTransferProxy is
+    Authorizable,
+    IAssetTransferProxy
+{
+    // Mapping from Asset Proxy Id's to their respective Asset Proxy
+    mapping (uint8 => IAssetProxy) public assetProxies;
+
+    /// @dev Delegates transfer to the corresponding asset proxy.
+    /// @param assetMetadata Byte array encoded for the respective asset proxy.
+    /// @param from Address to transfer token from.
+    /// @param to Address to transfer token to.
+    /// @param amount Amount of token to transfer.
+    function transferFrom(
+        bytes assetMetadata,
+        address from,
+        address to,
+        uint256 amount)
+        public
+        onlyAuthorized
+    {
+        // Lookup asset proxy
+        require(assetMetadata.length >= 1);
+        uint8 assetProxyId = uint8(assetMetadata[0]);
+        IAssetProxy assetProxy = assetProxies[assetProxyId];
+        require(assetProxy != address(0x0));
+
+        // Delegate transfer to asset proxy
+        assetProxy.transferFrom(assetMetadata, from, to, amount);
+    }
+
+    /// @dev Registers a new asset proxy.
+    /// @param assetProxyId Id of the asset proxy.
+    /// @param newAssetProxyAddress Address of the asset proxy contract to register.
+    /// @param currentAssetProxyAddress Address of existing asset proxy to overwrite.
+    function registerAssetProxy(
+        uint8 assetProxyId,
+        address newAssetProxyAddress,
+        address currentAssetProxyAddress)
+        public
+        onlyAuthorized
+    {
+        // Ensure any existing asset proxy is not unintentionally overwritten
+        require(currentAssetProxyAddress == address(assetProxies[assetProxyId]));
+
+        // Ensure this method is not used to deregister asset proxies
+        require(newAssetProxyAddress != address(0x0));
+
+        // Store asset proxy and log registration
+        assetProxies[assetProxyId] = IAssetProxy(newAssetProxyAddress);
+        emit LogAssetProxyRegistration(assetProxyId, newAssetProxyAddress, currentAssetProxyAddress);
+    }
+
+    /// @dev Gets an asset proxy.
+    /// @param assetProxyId Id of the asset proxy.
+    /// @return The asset proxy registered to assetProxyId.
+    function getAssetProxy(uint8 assetProxyId)
+        public view
+        returns (IAssetProxy)
+    {
+        // Ensure asset proxy exists
+        IAssetProxy assetProxy = assetProxies[assetProxyId];
+        require(assetProxy != address(0x0));
+
+        // Return asset proxy
+        return assetProxy;
+    }
+
+    /// @dev Returns true if an asset proxy is registered.
+    /// @param assetProxyId Id of the asset proxy.
+    /// @return True if an asset proxy is registered. False, otherwise.
+    function isAssetProxyRegistered(uint8 assetProxyId)
```

Comment:
If we allow `getAssetProxy` to return 0, this function probably isn't necessary.